tests/java/verifiers/src/main/java/io/vectorized/reads_writes/Workload.java

```diff
@@ -0,0 +1,442 @@
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+public class Workload {
+  static enum OpStatus { WRITING, UNKNOWN, WRITTEN, SEEN, SKIPPED }
+
+  static volatile long started_s = -1;
+  static int workload_id_gen = 0;
+  static synchronized int get_workload_id() { return workload_id_gen++; }
+
+  static class OpInfo {
+    public long id;
+    public long started_us;
+    public OpStatus status;
+    public int partition;
+    public long offset;
+  }
+
+  static class WriterStat {
+    public long countWritten = 0;
+    public long countRead = 0;
+    public long lastOffset = -1;
+    public int inflights = 0;
+    public boolean draining = false;
+    public Semaphore drained = new Semaphore(0);
+  }
+
+  class SendCallback implements Callback {
+    private OpInfo op;
+    private WriterStat stat;
+
+    SendCallback(OpInfo op, WriterStat stat) {
+      this.op = op;
+      this.stat = stat;
+    }
+
+    public void onCompletion(RecordMetadata meta, Exception e) {
+      synchronized (Workload.this) {
+        if (e == null) {
+          log(op.partition, "d\t" + op.id + "\t" + meta.offset());
+          switch (op.status) {
+          case SKIPPED:
+          case UNKNOWN:
+          case WRITTEN:
+            violation(
+                op.partition,
+                "just written record can't have status: " + op.status.name());
+            break;
+          case WRITING:
+            op.status = OpStatus.WRITTEN;
+            op.offset = meta.offset();
+            break;
+          case SEEN:
+            if (op.offset != meta.offset()) {
+              violation(
+                  op.partition,
+                  "oid:" + op.id + " was seen with conflicting offsets; seen: "
+                      + op.offset + " written: " + meta.offset());
+              break;
+            }
+            break;
+          }
+
+          stat.countWritten++;
+          stat.lastOffset = Math.max(stat.lastOffset, op.offset);
+        } else {
+          log(op.partition, "e\t" + op.id);
+          System.out.println("=== Error on write");
+          System.out.println(e);
+          e.printStackTrace(System.out);
+          switch (op.status) {
+          case UNKNOWN:
+            Workload.this.violation(
+                op.partition,
+                "impossible situation: failed write can't already have unknown status");
+            break;
+          case WRITING:
+            op.status = OpStatus.UNKNOWN;
+            break;
+          default:
+            break;
+          }
+        }
+        stat.inflights--;
+        if (stat.draining && stat.inflights == 0) {
+          stat.drained.release();
+        }
+      }
+    }
+  }
+
+  private synchronized void log(int partition, String msg) {
```

Comment:
nit: would it be possible to organize this class using Java declaration ordering convention: 
```
According to [Code Conventions for the Java Programming Language](https://checkstyle.org/styleguides/sun-code-conventions-19990420/CodeConventions.doc2.html#a1852), the parts of a class or interface declaration should appear in the following order:

- Class (static) variables. First the public class variables, then protected, then package level (no access modifier), and then private.
- Instance variables. First the public class variables, then protected, then package level (no access modifier), and then private.
- Constructors
- Methods
```